weaviate/collections/queries/base.py

```diff
@@ -181,16 +181,18 @@ def __deserialize_non_ref_prop(self, value: properties_pb2.Value) -> Any:
             )
         if value.HasField("blob_value"):
             return value.blob_value
-        assert value.HasField("phone_value")
-        return _PhoneNumber(
-            country_code=value.phone_value.country_code,
-            default_country=value.phone_value.default_country,
-            international_formatted=value.phone_value.international_formatted,
-            national=value.phone_value.national,
-            national_formatted=value.phone_value.national_formatted,
-            number=value.phone_value.input,
-            valid=value.phone_value.valid,
-        )
+        if value.HasField("phone_value"):
+            return _PhoneNumber(
+                country_code=value.phone_value.country_code,
+                default_country=value.phone_value.default_country,
+                international_formatted=value.phone_value.international_formatted,
+                national=value.phone_value.national,
+                national_formatted=value.phone_value.national_formatted,
+                number=value.phone_value.input,
+                valid=value.phone_value.valid,
+            )
+        assert value.HasField("null_value")
+        return None
```

Comment:
Can we have another `if` here for the `null_value` field and then a final path that either throws a warning or an error saying that an unexpected field was detected and discarded (since the client doesn't know what to do with it)

Reply:
done!